File: app/src/main/java/com/lucidsoftworksllc/sabotcommunity/ChatroomMessagesAdapter.java
package com.lucidsoftworksllc.sabotcommunity;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import de.hdodenhof.circleimageview.CircleImageView;

public class ChatroomMessagesAdapter extends RecyclerView.Adapter<ChatroomMessagesAdapter.ViewHolder> {

    private String username;
    private Context context;
    private int SELF = 786;
    private ArrayList<ChatroomMessagesHelper> messages;
    public ChatroomMessagesAdapter(Context context, ArrayList<ChatroomMessagesHelper> messages, String username) {
        this.username = username;
        this.messages = messages;
        this.context = context;
    }

    @Override
    public int getItemViewType(int position) {
        ChatroomMessagesHelper message = messages.get(position);
        if (message.getUsername().equals(username)) {
            return SELF;
        }
        return position;
    }

    @NonNull
    @Override
    public ChatroomMessagesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView;
        if (viewType == SELF) {
            itemView = LayoutInflater.from(context)
                    .inflate(R.layout.recycler_chatroom_right, parent, false);
        } else {
            itemView = LayoutInflater.from(context)
                    .inflate(R.layout.recycler_chatroom_left, parent, false);
        }
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(ChatroomMessagesAdapter.ViewHolder holder, int position) {
        final ChatroomMessagesHelper message = messages.get(position);
        if (message.getVerified().equals("yes")){
            holder.verified.setVisibility(View.VISIBLE);
        }else{
            holder.verified.setVisibility(View.GONE);
        }
        if (message.getlast_online().equals("yes")){
            holder.online.setVisibility(View.VISIBLE);
        }else{
            holder.online.setVisibility(View.GONE);
        }
        String profile_pic = message.getProfile_pic().substring(0, message.getProfile_pic().length() - 4)+"_r.JPG";
        Glide.with(context)
                .load(Constants.BASE_URL + profile_pic)
                .into(holder.imageViewProfilenewsPic);
        holder.textViewBody.setText(message.getMessage());
        holder.profileCommentsDateTime_top.setText(message.getTime_message());
        holder.textViewProfileName.setText(message.getNickname());
        holder.textViewProfileName.setOnClickListener(v -> {
            FragmentProfile ldf = new FragmentProfile();
            Bundle args = new Bundle();
            args.putString("UserId", message.getUserid());
            ldf.setArguments(args);
            ((FragmentActivity) context).getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, ldf).addToBackStack(null).commit();
        });
    }

    @Override public int getItemCount() {
        return messages.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView textViewProfileName, textViewBody;
        TextView profileCommentsDateTime_top;
        CircleImageView imageViewProfilenewsPic, online, verified;
        public ViewHolder(View itemView) {
            super(itemView);
            textViewProfileName = itemView.findViewById(R.id.textViewProfileName);
            imageViewProfilenewsPic = itemView.findViewById(R.id.imageViewProfilenewsPic);
            online = itemView.findViewById(R.id.online);
            verified = itemView.findViewById(R.id.verified);
            profileCommentsDateTime_top = itemView.findViewById(R.id.profileCommentsDateTime_top);
            textViewBody = itemView.findViewById(R.id.textViewBody);
        }
    }

}

File: app/src/main/java/com/lucidsoftworksllc/sabotcommunity/adapters/ChatroomMessagesAdapter.kt
package com.lucidsoftworksllc.sabotcommunity.adapters

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.lucidsoftworksllc.sabotcommunity.models.ChatroomMessagesHelper
import com.lucidsoftworksllc.sabotcommunity.others.Constants
import com.lucidsoftworksllc.sabotcommunity.R
import com.lucidsoftworksllc.sabotcommunity.fragments.FragmentProfile
import de.hdodenhof.circleimageview.CircleImageView
import java.util.*

class ChatroomMessagesAdapter(private val context: Context, private val messages: ArrayList<ChatroomMessagesHelper>, private val username: String) : RecyclerView.Adapter<ChatroomMessagesAdapter.ViewHolder>() {
    private val SELF = 786
    override fun getItemViewType(position: Int): Int {
        val message = messages[position]
        return if (message.username == username) {
            SELF
        } else position
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView: View = if (viewType == SELF) {
            LayoutInflater.from(context)
                    .inflate(R.layout.recycler_chatroom_right, parent, false)
        } else {
            LayoutInflater.from(context)
                    .inflate(R.layout.recycler_chatroom_left, parent, false)
        }
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val message = messages[position]
        if (message.verified == "yes") {
            holder.verified.visibility = View.VISIBLE
        } else {
            holder.verified.visibility = View.GONE
        }
        if (message.lastonline == "yes") {
            holder.online.visibility = View.VISIBLE
        } else {
            holder.online.visibility = View.GONE
        }
        val profilePic = message.profile_pic.substring(0, message.profile_pic.length - 4) + "_r.JPG"
        Glide.with(context)
                .load(Constants.BASE_URL + profilePic)
                .into(holder.imageViewProfilenewsPic)
        holder.textViewBody.text = message.message
        holder.profileCommentsDateTimeTop.text = message.time_message
        holder.textViewProfileName.text = message.nickname
        holder.textViewProfileName.setOnClickListener {
            val ldf = FragmentProfile()
            val args = Bundle()
            args.putString("UserId", message.userid)
            ldf.arguments = args
            (context as FragmentActivity).supportFragmentManager.beginTransaction().setCustomAnimations(R.anim.slide_in, R.anim.fade_out, R.anim.fade_in, R.anim.slide_out).replace(R.id.fragment_container, ldf).addToBackStack(null).commit()
        }
    }

    override fun getItemCount(): Int {
        return messages.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textViewProfileName: TextView = itemView.findViewById(R.id.textViewProfileName)
        var textViewBody: TextView = itemView.findViewById(R.id.textViewBody)
        var profileCommentsDateTimeTop: TextView = itemView.findViewById(R.id.profileCommentsDateTime_top)
        var imageViewProfilenewsPic: CircleImageView = itemView.findViewById(R.id.imageViewProfilenewsPic)
        var online: CircleImageView = itemView.findViewById(R.id.online)
        var verified: CircleImageView = itemView.findViewById(R.id.verified)

    }
}